go/cli/internal/cli/install.go

```diff
@@ -38,20 +54,122 @@ func installChart(ctx context.Context, chartName string, namespace string, regis
+	// Build Helm values
+	helmProviderKey := GetModelProviderHelmValuesKey(modelProvider)
+	values := []string{
+		fmt.Sprintf("providers.default=%s", helmProviderKey),
+		fmt.Sprintf("providers.%s.apiKey=%s", helmProviderKey, apiKeyValue),
+	}
+
+	//allow user to set the helm registry and version
+	helmRegistry := GetEnvVarWithDefault(KAGENT_HELM_REPO, DefaultHelmOciRegistry)
+	helmVersion := GetEnvVarWithDefault(KAGENT_HELM_VERSION, version.Version)
+	helmExtraArgs := GetEnvVarWithDefault(KAGENT_HELM_EXTRA_ARGS, "")
+
+	// split helmExtraArgs by "--set" to get additional values
+	extraValues := strings.Split(helmExtraArgs, "--set")
+	for _, hev := range extraValues {
+		values = append(values, hev)
+	}
+
+	// Validate and normalize profile input
+	profile = strings.TrimSpace(profile)
+	switch profile {
+	case "":
+		// default to minimal. no warning as this is the default
+		profile = ProfileMinimal
+	case ProfileDemo, ProfileMinimal:
+		// valid, no change
+	default:
+		fmt.Fprintln(os.Stderr, "Invalid --profile value, defaulting to minimal")
+		profile = ProfileMinimal
+	}
+
+	// spinner for installation progress
+	s := spinner.New(spinner.CharSets[35], 100*time.Millisecond)
+
+	// First install kagent-crds
+	s.Suffix = " Installing kagent-crds from " + helmRegistry
+	defer s.Stop()
+	s.Start()
+	if output, err := installChart(ctx, "kagent-crds", cfg.Namespace, helmRegistry, helmVersion, nil, "", s); err != nil {
+		// Always stop the spinner before printing error messages
+		s.Stop()
 
+		// Check for various CRD existence scenarios, this is to be compatible with
+		// original kagent installation that had CRDs installed together with the kagent chart
+		if strings.Contains(output, "exists and cannot be imported into the current release") {
+			fmt.Fprintln(os.Stderr, "Warning: CRDs exist but aren't managed by helm.")
+			fmt.Fprintln(os.Stderr, "Run `uninstall` or delete them manually to")
+			fmt.Fprintln(os.Stderr, "ensure they're fully managed on next install.")
+			// Restart the spinner
+			s.Start()
+		} else {
+			fmt.Fprintln(os.Stderr, "Error installing kagent-crds:", output)
+			return nil
+		}
+	}
+
+	// Update status
+	s.Suffix = fmt.Sprintf(" Installing kagent with %s profile [%s] Using %s:%s %v", profile, modelProvider, helmRegistry, helmVersion, extraValues)
+	if output, err := installChart(ctx, "kagent", cfg.Namespace, helmRegistry, helmVersion, values, GetHelmProfileUrl(profile), s); err != nil {
+		// Always stop the spinner before printing error messages
+		s.Stop()
+		fmt.Fprintln(os.Stderr, "Error installing kagent:", output)
+		return nil
+	}
+
+	// Stop the spinner completely before printing the success message
+	s.Stop()
+	fmt.Fprintln(os.Stdout, "kagent installed successfully")
+
+	pf, err := NewPortForward(ctx, cfg)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error starting port-forward: %v\n", err)
+		return nil
+	}
+	return pf
+}
+
+func InteractiveInstallCmd(ctx context.Context, c *ishell.Context) *PortForward {
```

Comment:
Need to d.r.y. this.

I was thinking the non-interactive mode could use `--profile` flag, and interactive mode would have an interactive selection. so basically the entirety of the function is the same, aside from the signature and profile selection method.